client/app/components/composites/Menu/Menu.js

```diff
@@ -37,7 +37,7 @@ class Menu extends Component {
     const menuLabel = ReactDOM.findDOMNode(this.menuLabel);
 
     this.setState({ // eslint-disable-line react/no-did-mount-set-state, react/no-set-state
-      arrowPosition: menuLabel.offsetWidth / 2, // eslint-disable-line no-magic-numbers
+      arrowPosition: menuLabel.offsetWidth > 40 ? menuLabel.offsetWidth / 2 : 20, // eslint-disable-line no-magic-numbers
```

Comment:
This `componentDidMount` + `setState` + `offsetWidth` concerns me. This makes the component render twice initially, with the `offsetWidth` check causing an additional Layout Trashing in the middle. But I don't have a better solution for this.

What are the magic numbers 40 and 20?